lib/src/states/grid_movies_states.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+import '../utils/colors_movies.dart';
+import '../state_type.dart';
+import '../utils/ui_constants.dart';
+import 'movies_inkwell.dart';
+
+class GridMoviesStates extends StatelessWidget {
+  final StateType state;
+  final int itemCount;
+
+  const GridMoviesStates({
+    required this.state,
+    required this.itemCount,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      child: GridView.builder(
+        shrinkWrap: true,
+        physics: ScrollPhysics(),
+        itemCount: itemCount,
+        gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+          mainAxisExtent: UiConstants.mainAxisExtent,
+          crossAxisCount: UiConstants.gridColumns,
+        ),
+        itemBuilder: (
+          BuildContext context,
+          int index,
+        ) {
+          return GridTile(
+            child: Container(
+              margin: EdgeInsets.all(
+                UiConstants.gridTileMargin,
+              ),
+              child: Container(
+                child: ClipRRect(
+                  borderRadius: BorderRadius.only(
+                    topLeft: Radius.circular(
+                      UiConstants.gridCardsBorderRadius,
+                    ),
+                    bottomRight: Radius.circular(
+                      UiConstants.gridCardsBorderRadius,
+                    ),
+                  ),
+                  child: state.stateType == MoviesStateType.success
+                      ? MoviesInkwell(
+                          moviePosterPath:
+                              state.movies!.results![index].posterPath!,
+                          movieOverview:
+                              state.movies!.results![index].overView!,
+                          movieTitle: state.movies!.results![index].title,
+                          movieReleaseDate:
+                              state.movies!.results![index].releaseDate,
+                          movieVoteAverage: state
+                              .movies!.results![index].voteAverage
+                              .toString(),
+                        )
+                      : Center(
+                          child: CircularProgressIndicator(
+                            strokeWidth:
+                                UiConstants.progressIndicatorStrokeWidth,
+                            backgroundColor: ColorsMovies.colorARGB,
+                            valueColor: AlwaysStoppedAnimation<Color>(
+                              Colors.black,
+                            ),
+                          ),
+                        ),
```

Comment:
I think this is the same as CardSwiperStateLoading widget

Reply:
fixed